src/OpenSSL/crypto.py

```diff
@@ -1208,10 +1209,12 @@ def has_expired(self):
             :py:const:`False` otherwise.
         :rtype: :py:class:`bool`
         """
-        now = int(time())
-        notAfter = _lib.X509_get_notAfter(self._x509)
-        return _lib.ASN1_UTCTIME_cmp_time_t(
-            _ffi.cast('ASN1_UTCTIME*', notAfter), now) < 0
+        time_string = self.get_notAfter()
```

Comment:
you need to wrap this with `_native()` (which is an alias for `_util.native()`).